app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -86,7 +86,9 @@ class MainActivity : AppCompatActivity(), ApplicationContract.View {
     }
 
     override fun populateDeparturesTable(departuresList: List<DepartureInformation>) {
-        departureTimes.add(departuresList[0].arrivalTime)
+        departures.clear()
```

Comment:
If you only ever completely replace the contents of departures, is there any reason to use a mutableList, as opposed to just a List object that you can replace completely in this function?

Reply:
Not using a mutable list seems to cause issues with the recycler view

Reply:
Weird, even if you're updating it in both MainActivity and TableAdapter?

Reply:
I think so. It doesn't cause any errors in build or while running but the recycler view never repopulates (when the button is pressed) after initially being empty.